v2/tools/generator/internal/config/type_transformer.go

```diff
@@ -316,6 +316,12 @@ func (transformer *TypeTransformer) transformProperties(
 				return nil, eris.Wrapf(err, "transforming property %s", propertyName)
 			}
 
+			if transformer.Target.Optional {
```

Comment:
Should we error if somebody sets both option and required to true?

Reply:
Good idea. Done.